flake.nix

```diff
@@ -12,32 +12,40 @@
       overlays.default = final: prev:
         let
           packages = self.packages.${final.system};
-          inherit (packages) node-packages;
+          inherit (packages) nodePackages;
         in
         {
           nodejs = final.nodejs_20;
           nodePackages = prev.nodePackages;
           yarn = (import unstableNixPkgs { system = final.system; }).yarn-berry;
         };
     in
-    flake-utils.lib.eachDefaultSystem (system: {
-      devShell =
-        let pkgs = import nixpkgs {
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        pkgs = import nixpkgs {
           inherit system;
-
           overlays = [
             devshell.overlays.default
             overlays.default
           ];
         };
-        in
-        pkgs.devshell.mkShell {
+      in
+      {
+        devShell = pkgs.devshell.mkShell {
           packages = with pkgs; [
             nixpkgs-fmt
             nodejs_20
             yarn
             docker-compose
           ];
+          devshell = {
+            startup = {
+              setup.text = ''
+                export PATH="$PWD/node_modules/.bin:$PWD/packages/daemon/node_modules/.bin:$PWD/packages/wallet-service/node_modules/.bin:$PATH"
```

Comment:
question(non-blocking): While I think adding the `.bin` folder to the path is a smart move, maybe it relies too much on the internal implementation of NodeJS.

What is our actual need for this export? Couldn't we just replace our usage of those binaries with `npx myDependency`, since `npx` is available [since npm 8](https://docs.npmjs.com/cli/v8/commands/npx)?

Reply:
Yea, the idea here is to have `typescript` available in the path, so the LSP can use it

This it good so we don't have to install `typescript` globally (using npm)

Also, we are using `yarn` in this repository, so we shouldn't use `npm` here